{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **Generating Taxonomic Profile from Microbiome Data**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **1. Generating artifact**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mImported manifest.tsv as SingleEndFastqManifestPhred33V2 to sequences.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "##### FASTAQ to Artifact #####\n",
    "# import the fasta files into a QIIME2 artifact\n",
    "!qiime tools import \\\n",
    "  --type 'SampleData[SequencesWithQuality]' \\\n",
    "  --input-format SingleEndFastqManifestPhred33V2 \\\n",
    "  --input-path manifest.tsv \\\n",
    "  --output-path sequences.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: qualities.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "##### Artifact to Visualization #####\n",
    "# QIIME to visualize our sequencing data.\n",
    "!qiime demux summarize \\\n",
    "\t--i-data sequences.qza \\\n",
    "\t--o-visualization qualities.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting.Opening in existing browser session.\n",
      "\n",
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting."
     ]
    }
   ],
   "source": [
    "# visualize\n",
    "!qiime tools view qualities.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **2. Denoise/Demultiplex data**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: run_dada.R --input_directory /tmp/qiime2/davo/data/356c47dc-2fe3-48f9-a044-3371720cda23/data --output_path /tmp/tmpbptgo5n3/output.tsv.biom --output_track /tmp/tmpbptgo5n3/track.tsv --filtered_directory /tmp/tmpbptgo5n3 --truncation_length 250 --trim_left 0 --max_expected_errors 2.0 --truncation_quality_score 2 --max_length Inf --pooling_method independent --chimera_method consensus --min_parental_fold 1.0 --allow_one_off False --num_threads 4 --learn_min_reads 1000000 --homopolymer_gap_penalty NULL --band_size 16\n",
      "\n",
      "R version 4.3.3 (2024-02-29) \n",
      "Loading required package: Rcpp\n",
      "\u001b[?25hDADA2: 1.30.0 / Rcpp: 1.0.13.1 / RcppParallel: 5.1.9 \n",
      "\u001b[?25h\u001b[?25h2) Filtering \u001b[?25h\u001b[?25h..................................................................................................................................................................................\u001b[?25h\u001b[?25h\u001b[?25h\n",
      "\u001b[?25h\u001b[?25h3) Learning Error Rates\n",
      "\u001b[?25h259935750 total bases in 1039743 reads from 53 samples will be used for learning the error rates.\n",
      "\u001b[?25h4) Denoise samples \n",
      "..................................................................................................................................................................................\n",
      "\u001b[?25h5) Remove chimeras (method = consensus)\n",
      "\u001b[?25h\u001b[?25h6) Report read numbers through the pipeline\n",
      "\u001b[?25h\u001b[?25h7) Write output\n",
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[32mSaved FeatureTable[Frequency] to: dada/table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: dada/representative_sequences.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[DADA2Stats] to: dada/denoising_stats.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "##### Quality Filtering: From Sequence to ASV ##### Change the trunc-len to 250\n",
    "!qiime dada2 denoise-single \\\n",
    "    --i-demultiplexed-seqs sequences.qza \\\n",
    "    --p-trunc-len 250 \\\n",
    "    --p-n-threads 4 \\\n",
    "    --output-dir dada --verbose"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **3. Generating OTU**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: denoising-stats.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "# Denoising statistics\n",
    "!qiime metadata tabulate \\\n",
    "    --m-input-file dada/denoising_stats.qza \\\n",
    "    --o-visualization denoising-stats.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ./dada_freqtable.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "# Feature table summary\n",
    "!qiime feature-table summarize \\\n",
    "  --i-table ./dada/table.qza \\\n",
    "  --m-sample-metadata-file ./metadata.tsv \\\n",
    "  --o-visualization ./dada_freqtable.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting.Opening in existing browser session.\n",
      "\n",
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting."
     ]
    }
   ],
   "source": [
    "# visualize \n",
    "!qiime tools view ./dada_freqtable.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **4. Taxonomic Classification (math features to labels)**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-12-08 05:50:58 URL:https://s3-us-west-2.amazonaws.com/qiime2-data/classifiers/sklearn-1.4.2/silva/silva-138-99-nb-classifier.qza [218245868/218245868] -> \"silva-138-99-515-806-nb-classifier.qza\" [1]\n"
     ]
    }
   ],
   "source": [
    "# get the classifier\n",
    "!wget -nv -O \\\n",
    "  \"silva-138-99-nb-classifier.qza\" \\\n",
    "  \"https://data.qiime2.org/classifiers/sklearn-1.4.2/silva/silva-138-99-nb-classifier.qza\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mExported dada/table.qza as BIOMV210DirFmt to directory exported_table\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "# expoert the table\n",
    "!qiime tools export \\\n",
    "  --input-path dada/table.qza \\\n",
    "  --output-path exported_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert biom into csv\n",
    "# !biom convert \\\n",
    "# \t-i exported_table/feature-table.biom \\\n",
    "# \t-o exported_table/otu_table.csv --to-tsv\n",
    "\n",
    "## TRICKY NOT NECESSARY"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[Taxonomy] to: taxonomy.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "# Get taxonomic OTU\n",
    "!qiime feature-classifier classify-sklearn \\\n",
    "  --i-classifier silva-138-99-nb-classifier.qza \\\n",
    "  --i-reads ./dada/representative_sequences.qza \\\n",
    "  --o-classification taxonomy.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert biom into csv\n",
    "!biom convert \\\n",
    "\t-i exported_table/feature-table.biom \\\n",
    "\t-o exported_table/otu_table.csv --to-tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extract taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: collapsed_family_table.qza\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime taxa collapse \\\n",
    "  --i-table ./dada/table.qza  \\\n",
    "  --i-taxonomy taxonomy.qza \\\n",
    "  --p-level 6 \\\n",
    "  --o-collapsed-table collapsed_family_table.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mExported collapsed_family_table.qza as BIOMV210DirFmt to directory exported_family_table\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime tools export \\\n",
    "  --input-path collapsed_family_table.qza \\\n",
    "  --output-path exported_family_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [],
   "source": [
    "# visualize\n",
    "import biom\n",
    "import pandas as pd\n",
    "\n",
    "taxon = biom.load_table('exported_family_table/feature-table.biom')\n",
    "taxon = taxon.to_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>206534</th>\n",
       "      <th>206536</th>\n",
       "      <th>206538</th>\n",
       "      <th>206547</th>\n",
       "      <th>206548</th>\n",
       "      <th>206561</th>\n",
       "      <th>206562</th>\n",
       "      <th>206563</th>\n",
       "      <th>206564</th>\n",
       "      <th>206569</th>\n",
       "      <th>...</th>\n",
       "      <th>222171</th>\n",
       "      <th>224323</th>\n",
       "      <th>224324</th>\n",
       "      <th>224325</th>\n",
       "      <th>224326</th>\n",
       "      <th>224327</th>\n",
       "      <th>224328</th>\n",
       "      <th>224330</th>\n",
       "      <th>224844</th>\n",
       "      <th>224845</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides</th>\n",
       "      <td>13855.0</td>\n",
       "      <td>12252.0</td>\n",
       "      <td>5345.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>3807.0</td>\n",
       "      <td>5954.0</td>\n",
       "      <td>5657.0</td>\n",
       "      <td>1060.0</td>\n",
       "      <td>8001.0</td>\n",
       "      <td>...</td>\n",
       "      <td>12555.0</td>\n",
       "      <td>18068.0</td>\n",
       "      <td>27582.0</td>\n",
       "      <td>16909.0</td>\n",
       "      <td>14085.0</td>\n",
       "      <td>20334.0</td>\n",
       "      <td>23465.0</td>\n",
       "      <td>19140.0</td>\n",
       "      <td>5663.0</td>\n",
       "      <td>11811.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d__Bacteria;p__Firmicutes;c__Clostridia;o__Oscillospirales;f__Ruminococcaceae;g__Faecalibacterium</th>\n",
       "      <td>652.0</td>\n",
       "      <td>570.0</td>\n",
       "      <td>201.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>2887.0</td>\n",
       "      <td>6126.0</td>\n",
       "      <td>7048.0</td>\n",
       "      <td>1276.0</td>\n",
       "      <td>3372.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2197.0</td>\n",
       "      <td>6179.0</td>\n",
       "      <td>4676.0</td>\n",
       "      <td>5388.0</td>\n",
       "      <td>6230.0</td>\n",
       "      <td>8292.0</td>\n",
       "      <td>7873.0</td>\n",
       "      <td>3787.0</td>\n",
       "      <td>6680.0</td>\n",
       "      <td>14629.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Enterobacterales;f__Enterobacteriaceae;g__Escherichia-Shigella</th>\n",
       "      <td>1713.0</td>\n",
       "      <td>1871.0</td>\n",
       "      <td>1022.0</td>\n",
       "      <td>7127.0</td>\n",
       "      <td>2599.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>139.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>34.0</td>\n",
       "      <td>1926.0</td>\n",
       "      <td>2501.0</td>\n",
       "      <td>2335.0</td>\n",
       "      <td>2555.0</td>\n",
       "      <td>4727.0</td>\n",
       "      <td>4011.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>547.0</td>\n",
       "      <td>299.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d__Bacteria;p__Firmicutes;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__[Ruminococcus]_torques_group</th>\n",
       "      <td>2149.0</td>\n",
       "      <td>1203.0</td>\n",
       "      <td>1052.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>120.0</td>\n",
       "      <td>...</td>\n",
       "      <td>84.0</td>\n",
       "      <td>1881.0</td>\n",
       "      <td>1230.0</td>\n",
       "      <td>500.0</td>\n",
       "      <td>680.0</td>\n",
       "      <td>1227.0</td>\n",
       "      <td>1708.0</td>\n",
       "      <td>182.0</td>\n",
       "      <td>86.0</td>\n",
       "      <td>142.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d__Bacteria;p__Firmicutes;c__Clostridia;o__Lachnospirales;f__Lachnospiraceae;g__[Ruminococcus]_gnavus_group</th>\n",
       "      <td>30.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>1270.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>134.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1200.0</td>\n",
       "      <td>981.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d__Bacteria;p__Proteobacteria;c__Alphaproteobacteria;o__Rhizobiales;f__Rhizobiaceae;g__Ochrobactrum</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Caloramatoraceae;__</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d__Bacteria;p__Proteobacteria;c__Gammaproteobacteria;o__Burkholderiales;f__Rhodocyclaceae;__</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d__Bacteria;p__Firmicutes;c__Clostridia;o__Oscillospirales;f__Hungateiclostridiaceae;g__Saccharofermentans</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>d__Bacteria;p__Firmicutes;c__Clostridia;o__Peptostreptococcales-Tissierellales;f__Family_XI;g__Gallicola</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>710 rows × 178 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     206534   206536  206538  \\\n",
       "d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__B...  13855.0  12252.0  5345.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...    652.0    570.0   201.0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...   1713.0   1871.0  1022.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...   2149.0   1203.0  1052.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...     30.0     40.0    17.0   \n",
       "...                                                     ...      ...     ...   \n",
       "d__Bacteria;p__Proteobacteria;c__Alphaproteobac...        0        0       0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Clos...        0        0       0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...        0        0       0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...        0        0       0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Pept...        0        0       0   \n",
       "\n",
       "                                                    206547  206548  206561  \\\n",
       "d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__B...     4.0    27.0  3807.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...     5.0    23.0  2887.0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...  7127.0  2599.0    25.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...    43.0    19.0       0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...  1270.0   124.0       0   \n",
       "...                                                    ...     ...     ...   \n",
       "d__Bacteria;p__Proteobacteria;c__Alphaproteobac...       0       0       0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Clos...       0       0       0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...       0       0       0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...       0       0       0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Pept...       0       0       0   \n",
       "\n",
       "                                                    206562  206563  206564  \\\n",
       "d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__B...  5954.0  5657.0  1060.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...  6126.0  7048.0  1276.0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...    43.0    22.0   139.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...    15.0       0       0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...       0       0       0   \n",
       "...                                                    ...     ...     ...   \n",
       "d__Bacteria;p__Proteobacteria;c__Alphaproteobac...       0       0       0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Clos...       0       0       0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...       0       0       0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...       0       0       0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Pept...       0       0       0   \n",
       "\n",
       "                                                    206569  ...   222171  \\\n",
       "d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__B...  8001.0  ...  12555.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...  3372.0  ...   2197.0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...     2.0  ...     34.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...   120.0  ...     84.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...       0  ...        0   \n",
       "...                                                    ...  ...      ...   \n",
       "d__Bacteria;p__Proteobacteria;c__Alphaproteobac...       0  ...        0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Clos...       0  ...        0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...       0  ...        0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...       0  ...        0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Pept...       0  ...        0   \n",
       "\n",
       "                                                     224323   224324   224325  \\\n",
       "d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__B...  18068.0  27582.0  16909.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...   6179.0   4676.0   5388.0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...   1926.0   2501.0   2335.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...   1881.0   1230.0    500.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...    134.0     61.0        0   \n",
       "...                                                     ...      ...      ...   \n",
       "d__Bacteria;p__Proteobacteria;c__Alphaproteobac...        0        0        0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Clos...        0        0        0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...        0        0        0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...        0      2.0        0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Pept...        0      2.0        0   \n",
       "\n",
       "                                                     224326   224327   224328  \\\n",
       "d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__B...  14085.0  20334.0  23465.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...   6230.0   8292.0   7873.0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...   2555.0   4727.0   4011.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...    680.0   1227.0   1708.0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...        0   1200.0    981.0   \n",
       "...                                                     ...      ...      ...   \n",
       "d__Bacteria;p__Proteobacteria;c__Alphaproteobac...        0        0        0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Clos...        0        0        0   \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...        0        0        0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...        0        0        0   \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Pept...        0        0        0   \n",
       "\n",
       "                                                     224330  224844   224845  \n",
       "d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__B...  19140.0  5663.0  11811.0  \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...   3787.0  6680.0  14629.0  \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...     99.0   547.0    299.0  \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...    182.0    86.0    142.0  \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Lach...    187.0       0        0  \n",
       "...                                                     ...     ...      ...  \n",
       "d__Bacteria;p__Proteobacteria;c__Alphaproteobac...        0       0        0  \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Clos...        0       0        0  \n",
       "d__Bacteria;p__Proteobacteria;c__Gammaproteobac...        0       0      2.0  \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Osci...        0       0        0  \n",
       "d__Bacteria;p__Firmicutes;c__Clostridia;o__Pept...        0       0        0  \n",
       "\n",
       "[710 rows x 178 columns]"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "taxon"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>c__Bacteroidia</td>\n",
       "      <td>o__Bacteroidales</td>\n",
       "      <td>f__Bacteroidaceae</td>\n",
       "      <td>g__Bacteroides</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Clostridia</td>\n",
       "      <td>o__Oscillospirales</td>\n",
       "      <td>f__Ruminococcaceae</td>\n",
       "      <td>g__Faecalibacterium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>c__Gammaproteobacteria</td>\n",
       "      <td>o__Enterobacterales</td>\n",
       "      <td>f__Enterobacteriaceae</td>\n",
       "      <td>g__Escherichia-Shigella</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Clostridia</td>\n",
       "      <td>o__Lachnospirales</td>\n",
       "      <td>f__Lachnospiraceae</td>\n",
       "      <td>g__[Ruminococcus]_torques_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Clostridia</td>\n",
       "      <td>o__Lachnospirales</td>\n",
       "      <td>f__Lachnospiraceae</td>\n",
       "      <td>g__[Ruminococcus]_gnavus_group</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>705</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>c__Alphaproteobacteria</td>\n",
       "      <td>o__Rhizobiales</td>\n",
       "      <td>f__Rhizobiaceae</td>\n",
       "      <td>g__Ochrobactrum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>706</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Clostridia</td>\n",
       "      <td>o__Clostridiales</td>\n",
       "      <td>f__Caloramatoraceae</td>\n",
       "      <td>__</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>707</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>c__Gammaproteobacteria</td>\n",
       "      <td>o__Burkholderiales</td>\n",
       "      <td>f__Rhodocyclaceae</td>\n",
       "      <td>__</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>708</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Clostridia</td>\n",
       "      <td>o__Oscillospirales</td>\n",
       "      <td>f__Hungateiclostridiaceae</td>\n",
       "      <td>g__Saccharofermentans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>709</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Clostridia</td>\n",
       "      <td>o__Peptostreptococcales-Tissierellales</td>\n",
       "      <td>f__Family_XI</td>\n",
       "      <td>g__Gallicola</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>710 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               0                  1                       2  \\\n",
       "0    d__Bacteria    p__Bacteroidota          c__Bacteroidia   \n",
       "1    d__Bacteria      p__Firmicutes           c__Clostridia   \n",
       "2    d__Bacteria  p__Proteobacteria  c__Gammaproteobacteria   \n",
       "3    d__Bacteria      p__Firmicutes           c__Clostridia   \n",
       "4    d__Bacteria      p__Firmicutes           c__Clostridia   \n",
       "..           ...                ...                     ...   \n",
       "705  d__Bacteria  p__Proteobacteria  c__Alphaproteobacteria   \n",
       "706  d__Bacteria      p__Firmicutes           c__Clostridia   \n",
       "707  d__Bacteria  p__Proteobacteria  c__Gammaproteobacteria   \n",
       "708  d__Bacteria      p__Firmicutes           c__Clostridia   \n",
       "709  d__Bacteria      p__Firmicutes           c__Clostridia   \n",
       "\n",
       "                                          3                          4  \\\n",
       "0                          o__Bacteroidales          f__Bacteroidaceae   \n",
       "1                        o__Oscillospirales         f__Ruminococcaceae   \n",
       "2                       o__Enterobacterales      f__Enterobacteriaceae   \n",
       "3                         o__Lachnospirales         f__Lachnospiraceae   \n",
       "4                         o__Lachnospirales         f__Lachnospiraceae   \n",
       "..                                      ...                        ...   \n",
       "705                          o__Rhizobiales            f__Rhizobiaceae   \n",
       "706                        o__Clostridiales        f__Caloramatoraceae   \n",
       "707                      o__Burkholderiales          f__Rhodocyclaceae   \n",
       "708                      o__Oscillospirales  f__Hungateiclostridiaceae   \n",
       "709  o__Peptostreptococcales-Tissierellales               f__Family_XI   \n",
       "\n",
       "                                   5  \n",
       "0                     g__Bacteroides  \n",
       "1                g__Faecalibacterium  \n",
       "2            g__Escherichia-Shigella  \n",
       "3    g__[Ruminococcus]_torques_group  \n",
       "4     g__[Ruminococcus]_gnavus_group  \n",
       "..                               ...  \n",
       "705                  g__Ochrobactrum  \n",
       "706                               __  \n",
       "707                               __  \n",
       "708            g__Saccharofermentans  \n",
       "709                     g__Gallicola  \n",
       "\n",
       "[710 rows x 6 columns]"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_split = taxon.index.str.split(';', expand=True).to_frame().reset_index(drop=True)\n",
    "df_split"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get table for model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Combine OTU and Taxonomy data\n",
    "import biom\n",
    "import pandas as pd\n",
    "\n",
    "# Load the OTU table\n",
    "otu = biom.load_table('exported_table/feature-table.biom')\n",
    "otu = otu.to_dataframe()\n",
    "\n",
    "# Load the taxonomy table\n",
    "taxonomy = pd.read_csv('exported_table/taxonomy.tsv', sep='\\t', index_col=0)\n",
    "\n",
    "# Merge the OTU table with the taxonomy table\n",
    "otu_taxonomy_merged = pd.merge(otu, taxonomy, left_index=True, right_index=True)\n",
    "\n",
    "# Save the merged table to a CSV file\n",
    "otu_taxonomy_merged.to_csv('otu_with_taxonomy.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count    5350.000000\n",
      "mean        0.948876\n",
      "std         0.079536\n",
      "min         0.315809\n",
      "25%         0.927707\n",
      "50%         0.992918\n",
      "75%         0.999658\n",
      "max         1.000000\n",
      "Name: Confidence, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "print(otu_taxonomy_merged['Confidence'].describe())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>206534</th>\n",
       "      <th>206536</th>\n",
       "      <th>206538</th>\n",
       "      <th>206547</th>\n",
       "      <th>206548</th>\n",
       "      <th>206561</th>\n",
       "      <th>206562</th>\n",
       "      <th>206563</th>\n",
       "      <th>206564</th>\n",
       "      <th>206569</th>\n",
       "      <th>...</th>\n",
       "      <th>224324</th>\n",
       "      <th>224325</th>\n",
       "      <th>224326</th>\n",
       "      <th>224327</th>\n",
       "      <th>224328</th>\n",
       "      <th>224330</th>\n",
       "      <th>224844</th>\n",
       "      <th>224845</th>\n",
       "      <th>Taxon</th>\n",
       "      <th>Confidence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>668fdb718997fc1589c7817655d4bb5f</th>\n",
       "      <td>6.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>1699.0</td>\n",
       "      <td>3325.0</td>\n",
       "      <td>3262.0</td>\n",
       "      <td>633.0</td>\n",
       "      <td>4924.0</td>\n",
       "      <td>...</td>\n",
       "      <td>12783.0</td>\n",
       "      <td>6685.0</td>\n",
       "      <td>5904.0</td>\n",
       "      <td>10805.0</td>\n",
       "      <td>12100.0</td>\n",
       "      <td>17885.0</td>\n",
       "      <td>1708.0</td>\n",
       "      <td>6589.0</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>0.999984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>a3f36ef32153f2fc2aaeac2feb23777f</th>\n",
       "      <td>482.0</td>\n",
       "      <td>509.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>750.0</td>\n",
       "      <td>2543.0</td>\n",
       "      <td>4525.0</td>\n",
       "      <td>767.0</td>\n",
       "      <td>742.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4024.0</td>\n",
       "      <td>1374.0</td>\n",
       "      <td>1739.0</td>\n",
       "      <td>6074.0</td>\n",
       "      <td>6076.0</td>\n",
       "      <td>2694.0</td>\n",
       "      <td>4684.0</td>\n",
       "      <td>9331.0</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Clostridia;o__Osc...</td>\n",
       "      <td>0.990108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9496d87b94d90dff068f0716603930bd</th>\n",
       "      <td>11952.0</td>\n",
       "      <td>10357.0</td>\n",
       "      <td>4574.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>455.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>6495.0</td>\n",
       "      <td>8555.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>0.999978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1b158b8b2922d4fcad5d9cea607cbb7d</th>\n",
       "      <td>1536.0</td>\n",
       "      <td>1648.0</td>\n",
       "      <td>805.0</td>\n",
       "      <td>7117.0</td>\n",
       "      <td>2599.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>139.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2495.0</td>\n",
       "      <td>2329.0</td>\n",
       "      <td>2483.0</td>\n",
       "      <td>4712.0</td>\n",
       "      <td>4008.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>547.0</td>\n",
       "      <td>299.0</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>0.908560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23fed68c6c76ab10ba1be8a43e9176e7</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>453.0</td>\n",
       "      <td>451.0</td>\n",
       "      <td>98.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>...</td>\n",
       "      <td>36.0</td>\n",
       "      <td>2154.0</td>\n",
       "      <td>2414.0</td>\n",
       "      <td>471.0</td>\n",
       "      <td>313.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>441.0</td>\n",
       "      <td>1034.0</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Clostridia;o__Osc...</td>\n",
       "      <td>0.996007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51946b237d59929b4f666bb5229fbacf</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Clostridia;o__Lac...</td>\n",
       "      <td>0.765569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2707ca30f01ee8e07aa645b208068852</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>0.984180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46b10e705d5fbdc5c8d8f3a24249591e</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>d__Bacteria;p__Firmicutes;c__Clostridia;o__Chr...</td>\n",
       "      <td>0.822727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4c5ce916f019b3ba5d0fa994d24aee1a</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>d__Bacteria;p__Proteobacteria;c__Gammaproteoba...</td>\n",
       "      <td>0.988518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>530d3f556e633849ac4580c76fdda317</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...</td>\n",
       "      <td>0.999795</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5350 rows × 180 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   206534   206536  206538  206547  206548  \\\n",
       "668fdb718997fc1589c7817655d4bb5f      6.0      9.0     4.0     4.0    19.0   \n",
       "a3f36ef32153f2fc2aaeac2feb23777f    482.0    509.0   159.0       0       0   \n",
       "9496d87b94d90dff068f0716603930bd  11952.0  10357.0  4574.0       0       0   \n",
       "1b158b8b2922d4fcad5d9cea607cbb7d   1536.0   1648.0   805.0  7117.0  2599.0   \n",
       "23fed68c6c76ab10ba1be8a43e9176e7        0        0       0     5.0       0   \n",
       "...                                   ...      ...     ...     ...     ...   \n",
       "51946b237d59929b4f666bb5229fbacf        0        0       0       0       0   \n",
       "2707ca30f01ee8e07aa645b208068852        0        0       0       0       0   \n",
       "46b10e705d5fbdc5c8d8f3a24249591e        0        0       0       0       0   \n",
       "4c5ce916f019b3ba5d0fa994d24aee1a        0        0       0       0       0   \n",
       "530d3f556e633849ac4580c76fdda317        0        0       0       0       0   \n",
       "\n",
       "                                  206561  206562  206563  206564  206569  ...  \\\n",
       "668fdb718997fc1589c7817655d4bb5f  1699.0  3325.0  3262.0   633.0  4924.0  ...   \n",
       "a3f36ef32153f2fc2aaeac2feb23777f   750.0  2543.0  4525.0   767.0   742.0  ...   \n",
       "9496d87b94d90dff068f0716603930bd    35.0    91.0   455.0   102.0    37.0  ...   \n",
       "1b158b8b2922d4fcad5d9cea607cbb7d    25.0    43.0    22.0   139.0     2.0  ...   \n",
       "23fed68c6c76ab10ba1be8a43e9176e7   173.0   453.0   451.0    98.0   214.0  ...   \n",
       "...                                  ...     ...     ...     ...     ...  ...   \n",
       "51946b237d59929b4f666bb5229fbacf       0       0       0       0       0  ...   \n",
       "2707ca30f01ee8e07aa645b208068852       0       0       0       0       0  ...   \n",
       "46b10e705d5fbdc5c8d8f3a24249591e       0       0       0       0       0  ...   \n",
       "4c5ce916f019b3ba5d0fa994d24aee1a       0       0       0       0       0  ...   \n",
       "530d3f556e633849ac4580c76fdda317       0       0       0       0       0  ...   \n",
       "\n",
       "                                   224324  224325  224326   224327   224328  \\\n",
       "668fdb718997fc1589c7817655d4bb5f  12783.0  6685.0  5904.0  10805.0  12100.0   \n",
       "a3f36ef32153f2fc2aaeac2feb23777f   4024.0  1374.0  1739.0   6074.0   6076.0   \n",
       "9496d87b94d90dff068f0716603930bd        0       0    10.0   6495.0   8555.0   \n",
       "1b158b8b2922d4fcad5d9cea607cbb7d   2495.0  2329.0  2483.0   4712.0   4008.0   \n",
       "23fed68c6c76ab10ba1be8a43e9176e7     36.0  2154.0  2414.0    471.0    313.0   \n",
       "...                                   ...     ...     ...      ...      ...   \n",
       "51946b237d59929b4f666bb5229fbacf        0       0       0        0        0   \n",
       "2707ca30f01ee8e07aa645b208068852        0       0       0        0        0   \n",
       "46b10e705d5fbdc5c8d8f3a24249591e        0       0       0        0        0   \n",
       "4c5ce916f019b3ba5d0fa994d24aee1a        0       0       0        0        0   \n",
       "530d3f556e633849ac4580c76fdda317        0       0       0        0        0   \n",
       "\n",
       "                                   224330  224844  224845  \\\n",
       "668fdb718997fc1589c7817655d4bb5f  17885.0  1708.0  6589.0   \n",
       "a3f36ef32153f2fc2aaeac2feb23777f   2694.0  4684.0  9331.0   \n",
       "9496d87b94d90dff068f0716603930bd        0       0       0   \n",
       "1b158b8b2922d4fcad5d9cea607cbb7d     99.0   547.0   299.0   \n",
       "23fed68c6c76ab10ba1be8a43e9176e7     13.0   441.0  1034.0   \n",
       "...                                   ...     ...     ...   \n",
       "51946b237d59929b4f666bb5229fbacf        0       0     2.0   \n",
       "2707ca30f01ee8e07aa645b208068852        0       0     2.0   \n",
       "46b10e705d5fbdc5c8d8f3a24249591e        0       0     2.0   \n",
       "4c5ce916f019b3ba5d0fa994d24aee1a        0       0     2.0   \n",
       "530d3f556e633849ac4580c76fdda317        0       0     2.0   \n",
       "\n",
       "                                                                              Taxon  \\\n",
       "668fdb718997fc1589c7817655d4bb5f  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...   \n",
       "a3f36ef32153f2fc2aaeac2feb23777f  d__Bacteria;p__Firmicutes;c__Clostridia;o__Osc...   \n",
       "9496d87b94d90dff068f0716603930bd  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...   \n",
       "1b158b8b2922d4fcad5d9cea607cbb7d  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "23fed68c6c76ab10ba1be8a43e9176e7  d__Bacteria;p__Firmicutes;c__Clostridia;o__Osc...   \n",
       "...                                                                             ...   \n",
       "51946b237d59929b4f666bb5229fbacf  d__Bacteria;p__Firmicutes;c__Clostridia;o__Lac...   \n",
       "2707ca30f01ee8e07aa645b208068852  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "46b10e705d5fbdc5c8d8f3a24249591e  d__Bacteria;p__Firmicutes;c__Clostridia;o__Chr...   \n",
       "4c5ce916f019b3ba5d0fa994d24aee1a  d__Bacteria;p__Proteobacteria;c__Gammaproteoba...   \n",
       "530d3f556e633849ac4580c76fdda317  d__Bacteria;p__Bacteroidota;c__Bacteroidia;o__...   \n",
       "\n",
       "                                  Confidence  \n",
       "668fdb718997fc1589c7817655d4bb5f    0.999984  \n",
       "a3f36ef32153f2fc2aaeac2feb23777f    0.990108  \n",
       "9496d87b94d90dff068f0716603930bd    0.999978  \n",
       "1b158b8b2922d4fcad5d9cea607cbb7d    0.908560  \n",
       "23fed68c6c76ab10ba1be8a43e9176e7    0.996007  \n",
       "...                                      ...  \n",
       "51946b237d59929b4f666bb5229fbacf    0.765569  \n",
       "2707ca30f01ee8e07aa645b208068852    0.984180  \n",
       "46b10e705d5fbdc5c8d8f3a24249591e    0.822727  \n",
       "4c5ce916f019b3ba5d0fa994d24aee1a    0.988518  \n",
       "530d3f556e633849ac4580c76fdda317    0.999795  \n",
       "\n",
       "[5350 rows x 180 columns]"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "otu_taxonomy_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>c__Bacteroidia</td>\n",
       "      <td>o__Bacteroidales</td>\n",
       "      <td>f__Bacteroidaceae</td>\n",
       "      <td>g__Bacteroides</td>\n",
       "      <td>s__</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Clostridia</td>\n",
       "      <td>o__Oscillospirales</td>\n",
       "      <td>f__Ruminococcaceae</td>\n",
       "      <td>g__Faecalibacterium</td>\n",
       "      <td>s__</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>c__Bacteroidia</td>\n",
       "      <td>o__Bacteroidales</td>\n",
       "      <td>f__Bacteroidaceae</td>\n",
       "      <td>g__Bacteroides</td>\n",
       "      <td>s__</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>c__Gammaproteobacteria</td>\n",
       "      <td>o__Enterobacterales</td>\n",
       "      <td>f__Enterobacteriaceae</td>\n",
       "      <td>g__Escherichia-Shigella</td>\n",
       "      <td>s__</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Clostridia</td>\n",
       "      <td>o__Oscillospirales</td>\n",
       "      <td>f__Ruminococcaceae</td>\n",
       "      <td>g__Faecalibacterium</td>\n",
       "      <td>s__</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5345</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Clostridia</td>\n",
       "      <td>o__Lachnospirales</td>\n",
       "      <td>f__Lachnospiraceae</td>\n",
       "      <td>g__Roseburia</td>\n",
       "      <td>s__</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5346</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>c__Gammaproteobacteria</td>\n",
       "      <td>o__Burkholderiales</td>\n",
       "      <td>f__Rhodocyclaceae</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5347</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Firmicutes</td>\n",
       "      <td>c__Clostridia</td>\n",
       "      <td>o__Christensenellales</td>\n",
       "      <td>f__Christensenellaceae</td>\n",
       "      <td>g__Christensenellaceae_R-7_group</td>\n",
       "      <td>s__</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5348</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Proteobacteria</td>\n",
       "      <td>c__Gammaproteobacteria</td>\n",
       "      <td>o__Burkholderiales</td>\n",
       "      <td>f__Sutterellaceae</td>\n",
       "      <td>g__Sutterella</td>\n",
       "      <td>s__</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5349</th>\n",
       "      <td>d__Bacteria</td>\n",
       "      <td>p__Bacteroidota</td>\n",
       "      <td>c__Bacteroidia</td>\n",
       "      <td>o__Bacteroidales</td>\n",
       "      <td>f__Rikenellaceae</td>\n",
       "      <td>g__Alistipes</td>\n",
       "      <td>s__</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5350 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                0                  1                       2  \\\n",
       "0     d__Bacteria    p__Bacteroidota          c__Bacteroidia   \n",
       "1     d__Bacteria      p__Firmicutes           c__Clostridia   \n",
       "2     d__Bacteria    p__Bacteroidota          c__Bacteroidia   \n",
       "3     d__Bacteria  p__Proteobacteria  c__Gammaproteobacteria   \n",
       "4     d__Bacteria      p__Firmicutes           c__Clostridia   \n",
       "...           ...                ...                     ...   \n",
       "5345  d__Bacteria      p__Firmicutes           c__Clostridia   \n",
       "5346  d__Bacteria  p__Proteobacteria  c__Gammaproteobacteria   \n",
       "5347  d__Bacteria      p__Firmicutes           c__Clostridia   \n",
       "5348  d__Bacteria  p__Proteobacteria  c__Gammaproteobacteria   \n",
       "5349  d__Bacteria    p__Bacteroidota          c__Bacteroidia   \n",
       "\n",
       "                          3                       4  \\\n",
       "0          o__Bacteroidales       f__Bacteroidaceae   \n",
       "1        o__Oscillospirales      f__Ruminococcaceae   \n",
       "2          o__Bacteroidales       f__Bacteroidaceae   \n",
       "3       o__Enterobacterales   f__Enterobacteriaceae   \n",
       "4        o__Oscillospirales      f__Ruminococcaceae   \n",
       "...                     ...                     ...   \n",
       "5345      o__Lachnospirales      f__Lachnospiraceae   \n",
       "5346     o__Burkholderiales       f__Rhodocyclaceae   \n",
       "5347  o__Christensenellales  f__Christensenellaceae   \n",
       "5348     o__Burkholderiales       f__Sutterellaceae   \n",
       "5349       o__Bacteroidales        f__Rikenellaceae   \n",
       "\n",
       "                                     5     6  \n",
       "0                       g__Bacteroides   s__  \n",
       "1                  g__Faecalibacterium   s__  \n",
       "2                       g__Bacteroides   s__  \n",
       "3              g__Escherichia-Shigella   s__  \n",
       "4                  g__Faecalibacterium   s__  \n",
       "...                                ...   ...  \n",
       "5345                      g__Roseburia   s__  \n",
       "5346                              None  None  \n",
       "5347  g__Christensenellaceae_R-7_group   s__  \n",
       "5348                     g__Sutterella   s__  \n",
       "5349                      g__Alistipes   s__  \n",
       "\n",
       "[5350 rows x 7 columns]"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_split = otu_taxonomy_merged.Taxon.str.split(';', expand=True).reset_index(drop=True)\n",
    "df_split"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **6. Alpha Rarefaction and Selecting a Rarefaction Depth**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: alpha_rarefaction_curves.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "# calculate rarefaction\n",
    "!qiime diversity alpha-rarefaction \\\n",
    "  --i-table ./dada/table.qza \\\n",
    "  --m-metadata-file metadata.tsv \\\n",
    "\t--p-min-depth 10 \\\n",
    "  --p-max-depth 4900 \\\n",
    "  --o-visualization alpha_rarefaction_curves.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting.Opening in existing browser session.\n",
      "\n",
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting."
     ]
    }
   ],
   "source": [
    "# visualize empress\n",
    "!qiime tools view alpha_rarefaction_curves.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## **3. Phylogenetics**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "^C\n",
      "\n",
      "Aborted!\n"
     ]
    }
   ],
   "source": [
    "# Aligning sequences and constructing a phylogenetic tree with QIIME2\n",
    "!qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "\t--i-sequences dada/representative_sequences.qza \\\n",
    "\t--output-dir tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: tree/empress.qzv\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "# Visualization for the tree using the empress QIIME 2 plugin\n",
    "!qiime empress tree-plot \\\n",
    "\t--i-tree tree/rooted_tree.qza \\\n",
    "\t--o-visualization tree/empress.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mExported taxonomy.qza as TSVTaxonomyDirectoryFormat to directory exported_table\u001b[0m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "# expoert the table\n",
    "!qiime tools export \\\n",
    "  --input-path taxonomy.qza \\\n",
    "  --output-path exported_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting.Opening in existing browser session.\n",
      "\n",
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting."
     ]
    }
   ],
   "source": [
    "# visualize empress\n",
    "!qiime tools view tree/empress.qzv"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "qiime2-amplicon",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
